src/Util/HTTPSig.php

```diff
@@ -0,0 +1,400 @@
+<?php
+
+/**
+ * @file src/Util/HTTPSig.php
+ */
+namespace Friendica\Util;
+
+use Friendica\Core\Config;
+use Friendica\Database\DBM;
+use Friendica\Util\Crypto;
+use Friendica\Util\HTTPHeaders;
+use dba;
+
+/**
+ * @brief Implements HTTP Signatures per draft-cavage-http-signatures-07.
+ *
+ * Ported from Hubzilla: https://framagit.org/hubzilla/core/blob/master/Zotlabs/Web/HTTPSig.php
+ * 
+ * @see https://tools.ietf.org/html/draft-cavage-http-signatures-07
+ */
+
+class HTTPSig
+{
+	/**
+	 * @brief RFC5843
+	 *
+	 * @see https://tools.ietf.org/html/rfc5843
+	 *
+	 * @param string  $body The value to create the digest for
+	 * @param boolean $set  (optional, default true)
+	 *   If set send a Digest HTTP header
+	 * 
+	 * @return string The generated digest of $body
+	 */
+	public static function generateDigest($body, $set = true)
+	{
+		$digest = base64_encode(hash('sha256', $body, true));
+
+		if($set) {
+			header('Digest: SHA-256=' . $digest);
```

Comment:
Nope, nope nope nope. No header should be set in a function named `generateDigest`. It should be done in a less conspicuous place.